STM32F405RGTx/Core/Src/rtos_threads/measurements_thread.cpp

```diff
@@ -0,0 +1,82 @@
+#include <stdio.h>
+#include <stdint.h>
+#include "main.h"
+#include "measurements_thread.hpp"
+#include "mc.hpp"
+#include "cmsis_os.h"
+#include "threads.hpp"
+
+// TODO: add defines here for conversions..
+
+RTOSThread MeasurementsThread::thread;
+uint16_t MeasurementsThread::ADC_buffer[];
+
+void MeasurementsThread::initialize(){
+    thread = RTOSThread(
+        "measurements_thread",
+        1024,
+        osPriorityAboveNormal,
+        runMeasurements
+    );
+
+}
+
+void MeasurementsThread::HAL_ADC_ConvCpltCallback(ADC_HandleTypeDef* hadc) {
+    HAL_StatusTypeDef status = HAL_ADC_Stop_DMA(hadc);
+    if(status != HAL_OK) {
+        printf("Error: HAL_ADC_Stop_DMA failed with status code %d\r\n", status);
+        Error_Handler();
+    }
+    osThreadFlagsSet(thread.getThreadId(), 0x00000001U);        // set flag to signal that ADC conversion has completed
+}
```

Comment:
As Ryan mentioned in the C++ PR I think, this function is a hardware interrupt so it has be declared in the global scope instead of as part of the class